### PR TITLE
krel release-notes refactor & cleanup

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
This is is the final cleanup of the `krel release-notes` to clean up several months worth of changes we made to add the capability to create the pull requests to krel. It is a general refactor of `release-notes.go` removing duplicate code and simplifying things where possible.

According to what we discussed, this PR also simplifies the command line interface and limits the scope of krel to only create the pull requests for the notes of a given release. The rest of the tools and options will still be available in the general purpose `release-notes` tool.

The command line interface is now simpler. The GitHub organization of the user is now just `--org` and help messages are now more concise. Invoking `krel release-notes` without any options will now display the usage information instead of surprising the user running with some defaults.

To be able to re-generate the JSON notes of a given tag (and avoid bugs like #838 ), `addReferenceToAssetsFile()` is now idempotent.

#### Special notes for your reviewer:

`github.BranchExists()` tests are coming, I want to wait for #1344 to merge before making more changes to the github pkg

Docs will be updated once this merges to reflect changes to the command line interface

#### Does this PR introduce a user-facing change?

```release-note
* Function `BranchExists()` has been ported to the github package.
* The `addReferenceToAssetsFile()` function idempotent to enable re-generation of json files
* The GitHub organization flag to create PRs is now simply `--org` 
* The `--website-org` and `--draft-org` have been removed
* Fork checking is now done in an independent block which can be called independently
* `--create-draft-pr` and `--create-website-pr` can now be run at the same time
* Running krel release-notes without any option now shows usage info and exits
```

### CODE REVIEW

File: cmd/krel/cmd/release_notes.go [hunk 1]
```diff
@@ -80,21 +84,28 @@ permissions to your fork of k/sig-release and k-sigs/release-notes.`,
 		github.TokenEnvKey),
 	SilenceUsage:  true,
 	SilenceErrors: true,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		// If none of the operation modes is defined, show the usage help and exit
+		if !releaseNotesOpts.createDraftPR && !releaseNotesOpts.createWebsitePR {
+			if err := cmd.Help(); err != nil {
+				logrus.Error(err)
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:91]: Would this make more sense in terms of error handling?
```suggestion
				logrus.Fatal(err)
```
Otherwise we could use `PreRunE` and return the error.

File: cmd/krel/cmd/release_notes.go [hunk 2]
```diff
@@ -80,21 +84,28 @@ permissions to your fork of k/sig-release and k-sigs/release-notes.`,
...
 	RunE: func(cmd *cobra.Command, args []string) error {
+		// Run the PR creation function
 		return runReleaseNotes()
 	},
 }
 
 type releaseNotesOptions struct {
 	tag             string
-	draftOrg        string
+	GithubOrg       string
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:103]: What is the reason to have this one public?
```suggestion
	githubOrg       string
```

File: cmd/krel/cmd/release_notes.go [hunk 3]
```diff
@@ -115,60 +126,38 @@ func init() {
...
-		"the name of the fork of kuberntets-sigs/release-notes, the Release Notes Draft PR will be created from this repository",
+		"Update the Release Notes draft and create a PR in k/sig-release",
 	)
 
 	releaseNotesCmd.PersistentFlags().BoolVar(
 		&releaseNotesOpts.createWebsitePR,
 		"create-website-pr",
 		false,
-		"generate the Releas Notes to a local fork of relnotes.k8s.io and create a PR.  --draft-org and --draft-repo must be set along with this option",
+		"Patch the relnotes.k8s.io sources and generate a PR with the changes",
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:146]: I think we should start each help with lower or upper case (I think lower case is our default).

File: cmd/krel/cmd/release_notes.go [hunk 4]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
-	var result *releaseNotesResult
 
-	// Create the PR for relnotes.k8s.io
+	// before running the generators, verify that the repositories are ready
 	if releaseNotesOpts.createWebsitePR {
-		// Check cmd line options
-		if err = validateWebsitePROptions(); err != nil {
-			return errors.Wrap(err, "validating PR command line options")
+		if err = verifyFork(
+			fmt.Sprintf("%s%s", websiteBranchPrefix, tag),
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:193]:
```suggestion
			websiteBranchPrefix + tag,
```

File: cmd/krel/cmd/release_notes.go [hunk 5]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
+		); err != nil {
+			return errors.Wrapf(err, "while checking %s/%s fork", defaultKubernetesSigsOrg, defaultKubernetesSigsRepo)
 		}
+	}
 
-		// Verify the repository
-		if err = validateWebsitePRRepository(); err != nil {
+	if releaseNotesOpts.createDraftPR {
+		if err = verifyFork(
+			fmt.Sprintf("%s%s", draftBranchPrefix, tag),
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:203]:
```suggestion
			draftBranchPrefix + tag,
```

File: cmd/krel/cmd/release_notes.go [hunk 6]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
+	var websiteError, draftError error
 
+	// Create the PR for relnotes.k8s.io
+	if releaseNotesOpts.createWebsitePR {
 		// Run the website PR process
-		if err := createWebsitePR(tag, jsonStr); err != nil {
-			return errors.Wrapf(err, "generating release notes for tag %s", tag)
-		}
-		return nil
+		websiteError = createWebsitePR(tag)
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:216]: Is there any reason that we do not return earlier in case of an error, like:
```suggestion
		return errors.Wrap(createWebsitePR(tag), "creating website PR")
```

File: cmd/krel/cmd/release_notes.go [hunk 7]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
+	// verify the branch does not previously exist
+	branchExists, err := gh.BranchExists(
+		forkOwner, forkRepo, branchName,
+	)
+	if err != nil {
+		return errors.Wrap(err, "while checking if branch can be created")
+	}
+
+	if branchExists {
+		return fmt.Errorf(
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:278]:
```suggestion
		return errors.Errorf(
```

File: cmd/krel/cmd/release_notes.go [hunk 8]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
 	}
 
 	if !isrepo {
-		return errors.New(
-			fmt.Sprintf(
-				"Cannot create PR, %s/%s is not a fork of %s/%s",
-				releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo,
-				git.DefaultGithubOrg, git.DefaultGithubReleaseRepo,
-			),
+		return fmt.Errorf(
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:263]:
```suggestion
		return errors.Errorf(
```

File: cmd/krel/cmd/release_notes.go [hunk 9]
```diff
@@ -193,249 +182,158 @@ func runReleaseNotes() (err error) {
...
-// validateWebsitePRRepository checks if a repo is fit to create a PR
-func validateWebsitePRRepository() error {
+// createDraftPR pushes the release notes draft to the users fork
+func createDraftPR(tag string) (err error) {
+	s, err := util.TagStringToSemver(tag)
+	if err != nil {
+		return errors.Wrapf(err, "no valid tag: %v", tag)
+	}
+
+	start := util.SemverToTagString(semver.Version{
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:293]: Let’s add a comment why we use the previous first RC here.

File: cmd/krel/cmd/release_notes.go [hunk 10]
```diff
@@ -558,10 +453,13 @@ func addReferenceToAssetsFile(repoPath, newJSONFile string) error {
 	scanner := bufio.NewScanner(file)
 	var assetsBuffer bytes.Buffer
 	assetsFileWasModified := false
+	fileIsReferenced := false
 	for scanner.Scan() {
 		// Check if the assets file already has the json notes referenced:
 		if strings.Contains(scanner.Text(), fmt.Sprintf("assets/%s", newJSONFile)) {
-			return errors.New(fmt.Sprintf("assets.ts already has a reference to %s ", newJSONFile))
+			logrus.Warnf("file %s is already referenced in assets.ts", newJSONFile)
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:460]:
```suggestion
			logrus.Warnf("File %s is already referenced in assets.ts", newJSONFile)
```

File: cmd/krel/cmd/release_notes.go [hunk 11]
```diff
@@ -764,7 +668,7 @@ func releaseNotesJSON(tag string) (string, error) {
 	}
 
 	logrus.Infof("Using start tag %v", startTag)
-	logrus.Infof("Using end tag %v", tag)
+	logrus.Infof("Using end tag   %v", tag)
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:671]: Is this change intentional? If not, then I suggest:
```suggestion
	logrus.Infof("Using end tag %v", tag)
```

File: cmd/krel/cmd/release_notes.go [hunk 12]
```diff
@@ -818,7 +722,7 @@ func releaseNotesFrom(startTag string) (*releaseNotesResult, error) {
 	}
 
 	logrus.Infof("Using start tag %v", startTag)
-	logrus.Infof("Using end tag %v", releaseNotesOpts.tag)
+	logrus.Infof("Using end tag   %v", releaseNotesOpts.tag)
```
Comment (saschagrunert) [inline cmd/krel/cmd/release_notes.go:725]:
```suggestion
	logrus.Infof("Using end tag %v", releaseNotesOpts.tag)
```

File: cmd/krel/cmd/release_notes.go (hunk 9 above)
Comment (saschagrunert) [reply cmd/krel/cmd/release_notes.go:284]: Sounds good, thank you. 🙂

**Comment by puerco** [comment]
Thanks for the review sascha !
